src/operator/concat.cc

```diff
@@ -21,11 +21,9 @@ Operator* CreateOp<cpu>(ConcatParam param, int dtype) {
   if (1 == param.dim) {
     switch (dtype) {
       case mshadow::kFloat32:
-      op = new MKLConcatOp<cpu, float>(param);
-      break;
+      return new MKLConcatOp<cpu, float>(param);
```

Comment:
what's topology? You basically swap the forward and backward of concat to get slice